File: db4oj/test/src/com/db4o/test/TypedDerivedArray.java
/* Copyright (C) 2004   db4objects Inc.   http://www.db4o.com */

package com.db4o.test;

import com.db4o.*;

public class TypedDerivedArray {
	
	Atom[] atoms;
	
	public void store(){
		Test.deleteAllInstances(this);
		TypedDerivedArray tda = new TypedDerivedArray();
		Molecule[] mols = new Molecule[1];
		mols[0] = new Molecule("TypedDerivedArray"); 
		tda.atoms = mols;
		Test.store(tda);
	}
	
	public void test(){
		Test.forEach(new TypedDerivedArray(), new Visitor4() {
            public void visit(Object obj) {
            	TypedDerivedArray tda = (TypedDerivedArray)obj;
            	Test.ensure(tda.atoms instanceof Molecule[]);
            }
        });
	}
}

File: db4on/test/src/com/db4o/test/TypedDerivedArray.cs
/* Copyright (C) 2004   db4objects Inc.   http://www.db4o.com */

using System;
using com.db4o.foundation;
using j4o.lang;
namespace com.db4o.test 
{

	public class TypedDerivedArray 
	{
      
		public TypedDerivedArray() : base() 
		{
		}
		internal Atom[] atoms;
      
		public void Store() 
		{
			Tester.DeleteAllInstances(this);
			TypedDerivedArray tda1 = new TypedDerivedArray();
			Molecule[] mols1 = new Molecule[1];
			mols1[0] = new Molecule("TypedDerivedArray");
			tda1.atoms = mols1;
			Tester.Store(tda1);
		}
      
		public void Test() 
		{
			Tester.ForEach(new TypedDerivedArray(), new MyVisitor());
		}

		public class MyVisitor:Visitor4
		{
			public void Visit(Object obj) 
			{
				TypedDerivedArray tda1 = (TypedDerivedArray)obj;
				Tester.Ensure(tda1.atoms is Molecule[]);
			}
		}



	}
}